lib/Analysis/MaskAnalysis.cpp

```diff
@@ -315,8 +315,6 @@ LogicalResult MaskState::parseCmp(arith::CmpIOp cmpOp, const Location loc,
       cmpDim = i;
     }
   }
-  assert(cmpDim != -1 &&
```

Comment:
Even though removing this assert doesn't trigger any failures, I think the consequence is we end up not applying the mask at runtime. The code below this assert applies the mask to the dimension found in the loop. But since cmpDim == -1, no dimension will match, which is incorrect.

I think we need to understand why the assert is triggered only in the 1x1 case and change the algorithm accordingly. I'd just revert this change and create another PR for this fix since the two changes are not really related.

Reply:
sounds good, reverted this change, thank you!